esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -649,6 +649,17 @@ bool CFileSprayEx::onDFUWUSearch(IEspContext &context, IEspDFUWUSearchRequest &
     {
         context.ensureFeatureAccess(DFU_WU_URL, SecAccess_Read, ECLWATCH_DFU_WU_ACCESS_DENIED, "Access to DFU workunit is denied.");
 
+#ifdef _CONTAINERIZED
+        StringArray clusterNames;
+        Owned<IStringIterator> targets = getContainerTargetClusters(nullptr, nullptr);
```

Comment:
is it right to use getContainerTargetClusters() which also seems to get roxie services too?

I'm not clear on what this service method ('DFUWUSearch') is used for?
e.g. why us called DFUWUSearch but returns a list of targets?

Reply:
The DFUWUSearch retrieves the information for searching DFU WUs. It was built for the legacy ECLWatch page (and may be used in other places). The code here retrieves cluster names (used for searching the WUs by cluster).

Reply:
Agree the method was badly named.

Reply:
I think Jake is right though.  In the long run we may want a new section listing clusters rather than queues or services. 
 But to get this working as is for DFU purposes we want the roxie queues names.  Unfortunately we currently use the type name roxie to mean services but there are also queue types of roxie.  So we probably want to add a different type name to getContainerTargetClusters() to get the "roxie-services"... or maybe have a different function for those.
 
Btw a roxie cluster will show up once in the queue list, but might have multiple services pointing to it.

Reply:
Need to think some more on general that targets can be compileOnly (roxie), runnable (hthor, roxie-workunit, thor), or callable (roxie-services).  And we just need to look at where each type may be needed, and how to add support for that to the functions.. perhaps including the validateTarget function as well.

Reply:
In bare metal environment, any DFU WU can run in a roxie? In cloud environment, does it make sure to search DFU WUs by a target (thor target, roxie target, or hthor target)?

Reply:
The issue is that the roxie can expose multiple services and a queue and all the services show up in the config.

myroxie
   roxie-service1
   roxie-service2

When listing targets to send queries to we want roxie-service1, roxie-service2.
When listing targets to run workunits on we don't want any.  Generally we only want workunit (eclagent roxies).  Technically hat could change later if we have roxie listen on a queue.
For DFU I think we want myroxie.  This is the actual cluster and will be where the files are accessed. 

In the current way it's done that would be the queue of type roxie, not the roxie services.
In the long run we probably want to define clusters explicitly in the configMap.  Clearly showing the difference between clusters and queues... because otherwise different parts of the code could interpret things differently.